src/controllers/userController.js

```diff
@@ -0,0 +1,64 @@
+const User = require("../models/user")
+
+
+
+//Fetch User
+function fetchUsers(req, res){
+    User.find({}, (err, users) => {
+      if(err) {
+          return res.status(500).json({ message: err})
+      }else {
+          return res.status(200).json({ users })
+      }
+  })
+}
+  
+  
+
+
+//Fetch Single User
+ function fetchUserById(req, res){
+  let id = req.params.id
+  User.findById(id, (err, user) => {
+    if(err) {
+      return res.status(500).json({ message: err})
+  }
+  else if(!user){
+      return res.status(404).json({ message: "user not found"})
+  }
+  else {
+      return res.status(200).json({ user })
+  }
+
+  })
+}
+
+
+function fetchUsersProfile(req, res){
+    let { id } = req.user
+    let currentUser = {
```

Comment:
the essence of finding the user by id is so that you can fetch all of the user's information from the database